scripts/us_census/pep/monthly_population_estimate/preprocess.py

```diff
@@ -0,0 +1,482 @@
+import os
+import re
+import sys
+
+import numpy as np
+import pandas as pd
+from absl import app
+from absl import flags
+
+pd.set_option("display.max_columns", None)
+
+FLAGS = flags.FLAGS
+default_input_path = os.path.dirname(
+    os.path.abspath(__file__)) + os.sep + "input_data"
+
+if ("ip_path" in sys.argv and len(sys.argv) > 2):
+    default_input_path = sys.argv[2]
+flags.DEFINE_string("input_path", default_input_path, "Import Data File's List")
+
+
+def _extract_year(val: str) -> tuple:
+    """
+    This Methods returns true,year from the value contains year.
+    Otherwise false,''
+
+    Arguments:
+        val (str) : A string value contains data below format
+                    yyyy or yyyy [1] or .MM 1
+    Returns:
+        res (tuple) : Tuple with boolean value and year value or None
+    """
+    val = str(val).strip().split(' ', maxsplit=1)[0]
+    if val.isnumeric() and len(val) == 4:
+        return True, val
+    if val.isnumeric() and len(val) == 5:
+        return True, val[:4]
+    return False, None
+
+
+def _return_year(col: str) -> str:
+    """
+    This Methods returns year value if col contains year.
+    Otherwise pandas NA value.
+
+    Arguments:
+        col (str) : A string value contains data below format
+                    yyyy or yyyy [1] or .MM 1
+    Returns:
+        res (str) : String value with year yyyy or pandas NA value
+    """
+    res, out = _extract_year(col)
+    if res:
+        return out
+    return pd.NA
+
+
+def _return_month(col: str) -> str:
+    """
+    This Methods returns month and date value if col contains month, date.
+    Otherwise pandas NA value.
+
+    Arguments:
+        col (str) : A string value contains data below format
+                    yyyy or yyyy [1] or .MM 1
+    Returns:
+        res (str) : String value with month mm or pandas NA value
+    """
+    res = _extract_year(col)
+    if res[0]:
+        return pd.NA
+    return col
+
+
+def _sum_cols(col: pd.Series) -> pd.Series:
+    """
+    This method concats two DataFrame column values
+    with space in-between.
+
+    Arguments:
+        col[0] (Series) : DataFrame Column of dtype str
+        col[1] (Series) : DataFrame Column of dtype str
+
+    Returns:
+        res (Series) : Concatenated DataFrame Columns
+    """
+    res = col[0]
+    if col[1] is None:
+        return res
+    res = col[0] + ' ' + col[1]
+    return res
+
+
+def _year_range(col: pd.Series) -> str:
+    """
+    This method returns year range from the dataframe
+    column.
+
+    Arguments:
+        col (Series) : DataFrame Column of dtype str
+
+    Returns:
+        year_range (str) : String of Concatenated max and min year values
+    """
+    year_range = None
+    max_year = max(pd.to_datetime(col, errors='coerce').dt.year)
+    min_year = min(pd.to_datetime(col, errors='coerce').dt.year)
+    year_range = str(max_year) + '-' + str(min_year)
+    return year_range
+
+
+def _clean_csv_file(df: pd.DataFrame) -> pd.DataFrame:

```

Comment:
Is this the same function as the one in download.py.  Why do we do this here and there?

Reply:
Extra/Redundant Functions have been removed from preprocess.py